src/details/interpreter.hpp

```diff
@@ -84,6 +84,29 @@ struct Interpreter {
 
     auto visit_table_constructor(ast::Table table_constructor, Env& env) -> EvalResult;
 
+    /**
+     * Evaluates a list of expressions (like return values, function parameters, etc).
+     *
+     * The result contains a vallist with all the evaluated values. If the expression evaluate to
+     * a Vallist only the first value is taken (except for the last expression). If the last
+     * expression evaluates to a Vallist it will be appended.
+     *
+     * Example:
+     *
+     * Expressions evaluate to (where `{}` denotes a Vallist):
+     *
+     * ```
+     * {1, 2, 3}, "hi", nil, 5, {1, 2}, {7, 8, 9}
+     * ```
+     *
+     * The result will be:
+     *
+     * ```
+     * 1, "hi", nil, 5, 1, 7, 8, 9
```

Comment:
```suggestion
     * {42, 2, 3}, "hi", nil, 5, {22, 17}, {7, 8, 9}
     * ```
     *
     * The result will be:
     *
     * ```
     * 42, "hi", nil, 5, 22, 7, 8, 9
```

Reply:
ok, new formulation:
why is always the first value returned/chosen?